python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -53,30 +53,41 @@
 POST_DST_TRANSITION_SUFFIX = "_POST_DST_TRANSITION"
 
 
-def dst_safe_strftime(dt: datetime, timezone_str: str, fmt: str) -> str:
+def is_ambiguous_time(dt: datetime):
+    """Returns if a datetime is ambiguous due to DST transitions."""
+    tz = check.not_none(dt.tzinfo)
+    # UTC is never ambiguous
+    if tz.tzname == "UTC":
+        return False
+    offset_before = cast(
+        timedelta,
+        (tz.utcoffset(dt.replace(fold=0)) if dt.fold else tz.utcoffset(dt)),
+    )
+    offset_after = cast(
+        timedelta,
+        (tz.utcoffset(dt) if dt.fold else tz.utcoffset(dt.replace(fold=1))),
+    )
+    return offset_before > offset_after
+
+
+def dst_safe_strftime(dt: datetime, fmt: str, schedule_type: Optional[ScheduleType]) -> str:
     """A method for converting a datetime to a string which will append a suffix in cases where
     the resulting timestamp would be ambiguous due to DST transitions.
     """
     time_str = dt.strftime(fmt)
-    try:
-        create_pendulum_time(
-            dt.year,
-            dt.month,
-            dt.day,
-            dt.hour,
-            dt.minute,
-            dt.second,
-            dt.microsecond,
-            tz=timezone_str,
-            dst_rule=pendulum.TRANSITION_ERROR,
-        )
-    except pendulum.tz.exceptions.AmbiguousTime:  # type: ignore
-        if dt.fold == 1:
-            return time_str + POST_DST_TRANSITION_SUFFIX
+    if schedule_type != ScheduleType.HOURLY:
```

Comment:
I think schedule_type == ScheduleType.HOURLY is slightly different from "schedule repeats every hour". The latter includes esoteric cron strings like "*/15 * * * *" which could have this problem as well right?

Reply:
unfortunate -- I'll add in a test case and fix this up